[12.0][MIG] account_invoice_check_identical_invoice

vvrossem opened the pull request:
#### [Task](https://gestion.coopiteasy.be/web#id=4623&view_type=form&model=project.task&menu_id=338&action=479)

changes : 
- since in Odoo 12.0, the Date and Datetime fields are now handled in the ORM as date objects, no need to convert `dt.date.today()` into a `str`

robinkeunen committed:
[ADD] check for invoices with same partner, invoice date and amount total

robinkeunen committed:
add identical_invoice_confirmed field and refuse validation if identical_invoice_detected and confirmed not checked

robinkeunen committed:
add onchange function

robinkeunen committed:
cleanup

robinkeunen committed:
[ADD] confirmation checkbox on supplier invoices

robinkeunen committed:
[IMP] only check supplier invoices

robinkeunen committed:
[FIX] a_i_check_identical_invoice: operation on recordset

vvrossem committed:
[MIG] account_invoice_check_identical_invoice: 9.0 -> 10.0
- followed oca guidelines
- not tested on odoo 10.0

vvrossem committed:
[MIG] account_invoice_check_identical_invoice: 10.0 -> 11.0
- followed oca guidelines
- not tested on odoo 11.0

vvrossem committed:
[MIG] account_invoice_check_identical_invoice: 11.0 -> 12.0
- followed oca guidelines
- tested on odoo 12.0

vvrossem committed:
[IMP] account_invoice_check_identical_invoice: black formatter

vvrossem committed:
[IMP] account_invoice_check_identical_invoice: oca-gen-addon-readme

vvrossem committed:
[IMP] account_invoice_check_identical_invoice: xml indent

robinkeunen commented on account_invoice_check_identical_invoice/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+    "website": "https://coopiteasy.be",
+    "description": """
+        This module requires to check this box to validate the invoice 
+        if invoices with the same partner, invoice date and totam alount already
```
Move description to DESCRIPTION.rst and add key summary in the __manifest__. You can install pre-commit for theses checks.

Also here, `totam alount` should be `total amount`

vvrossem replied:
done

robinkeunen replied:
sorry, that was misleading. I'll setup pre-commit here next time my compulsive disorders hit.

robinkeunen commented on account_invoice_check_identical_invoice/models/account_invoice.py:
```diff
@@ -0,0 +1,78 @@
+    @api.multi
+    def _check_identical_invoice(self):
+        for invoice in self:
+            Invoice = self.env["account.invoice"]
```
You could put it out of the loop.

vvrossem replied:
done

robinkeunen commented on account_invoice_check_identical_invoice/models/account_invoice.py:
```diff
@@ -0,0 +1,78 @@
+                    and not invoice.date_invoice
+                )
+
+            duplicate_invoices = partner_invoices.filtered(
+                lambda i: equal_amount(i)
+                and (same_date(i) or invoiced_today(i))
+            )
```
good ol' Robin, did you notice the [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming))?

vvrossem replied:
sexy!

robinkeunen commented on account_invoice_check_identical_invoice/models/account_invoice.py:
```diff
@@ -0,0 +1,78 @@
+                and (same_date(i) or invoiced_today(i))
+            )
+
+            if duplicate_invoices:
+                invoice.identical_invoice_detected = True
+            else:
+                invoice.identical_invoice_detected = False
```
Could be
```py
invoice.identical_invoice_detected = bool(duplicate_invoices)
```

vvrossem replied:
done

robinkeunen reviewed (changes requested):
Small change needed to the __manifest__ and some suggestions. Otherwise 👌

vvrossem committed:
[IMP] account_invoice_check_identical_invoice: __manifest__ ...
- builtin bool() in place of if/else
- self.env["account.invoice"] out of for loop
